Sourcery/Parsing/FileParser.swift

```diff
@@ -344,20 +344,21 @@ extension FileParser {
     fileprivate func parseEnumAssociatedValues(_ body: String) -> [Enum.Case.AssociatedValue] {
         guard !body.isEmpty else { return [] }
 
-        /// name: type, otherType
-        let components = body.components(separatedBy: ",")
-        return components.enumerated().flatMap { idx, element in
-            let nameType = element.components(separatedBy: ":").map { $0.trimmingCharacters(in: .whitespacesAndNewlines) }
-
-            switch nameType.count {
-            case 1:
-                return Enum.Case.AssociatedValue(name: "\(idx)", typeName: nameType.first ?? "")
-            case 2:
-                return Enum.Case.AssociatedValue(name: nameType.first, typeName: nameType.last ?? "")
-            default:
-                print("\(logPrefix)parseEnumAssociatedValues: Unknown enum case body format \(body)")
-                return nil
-            }
+        let items = body.commaSeparated()
+        return items
+            .map({ $0.trimmingCharacters(in: .whitespaces) })
+            .enumerated()
+            .map {
+                let nameAndType = $1.colonSeparated().map({ $0.trimmingCharacters(in: .whitespaces) })
+                let defaultName: String? = $0 == 0 && items.count == 1 ? nil : "\($0)"
```

Comment:
In case of single associated value it will not have a name. Even though we can set it, but we can not access it when we bind it. In this case the name will be nil, otherwise if there is no name, or it is `_` we use index.

Btw I think we should change it, because though we can not access single associated value by name when binding, we must use it when creating an enum:

```swift
enum Some { case value(v: Int) }
if case let .value(a) = Some.value(v: 1) { }
```

It looks like it should be handled like method parameters names (argument label + parameter name)